File: 1874.py
import sys

N = int(sys.stdin.readline().rstrip())

cnt = 1
stack = []
op = []
valid = True

for i in range(N):
    current = int(sys.stdin.readline().rstrip())
    while cnt <= current:
        stack.append(cnt)
        op.append('+')
        cnt += 1
    
    if stack[-1] == current:
        stack.pop()
        op.append('-')
    else:
        valid = False
        break
    
if valid == True:
    for i in op:
        print(i)
else:
    print("NO")

File: 1874.cpp
# include <iostream>
# include <stack>
# include <queue>
using namespace std;

int main() {
    int N;
    cin >> N;
    cin.ignore();

    stack<int> temp;
    queue<char> op;
    int cnt = 1;
    bool valid = true;

    for (int i = 0; i < N; i++) {
        int current;
        cin >> current;
        cin.ignore();

        while (cnt <= current) {
            temp.push(cnt);
            op.push('+');
            cnt++;
        }

        if (temp.top() == current) {
            temp.pop();
            op.push('-');
        }
        else {
            valid = false;
            break;
        }
    }

    if (valid == true) {
        while (!op.empty()) {
            cout << op.front() << '\n';
            op.pop();
        }
    }
    else {
        cout << "NO";
    }
}